sympy/stats/rv.py

```diff
@@ -1217,7 +1234,29 @@ def sample_iter(expr, condition=None, size=(), library='scipy',
     if condition is not None:
         given_fn = lambdify(rvs, condition, **kwargs)
 
-    def return_generator():
+    def return_generator_infinite():
+        count = 0
+        np = import_module('numpy')
+        if np:
+            rand_state = np.random.default_rng(seed=seed)
+        else:
+            rand_state = None
```

Comment:
This is only needed for the special case of infinite `numsamples` which was supported in `1.6` and hence is extended for `1.7`.
However, `numpy` doesn't become a required dependency by this since, if numpy is not present in the system, it will just pass `None` as rand_state for continuing sampling for infinite time.
This cannot be applied to `pymc3` since, support for `pymc3` was introduced after `1.6` and only accepts `int` as seed and that is why an internal RNG cannot be created for sampling finite number of samples, as `pymc3` will not accept it.